app/assets/javascripts/discourse/app/components/glimmer-header/home-logo.gjs

```diff
@@ -0,0 +1,120 @@
+import Component from "@glimmer/component";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import concatClass from "discourse/helpers/concat-class";
+import { wantsNewWindow } from "discourse/lib/intercept-click";
+import DiscourseURL from "discourse/lib/url";
+import Session from "discourse/models/session";
+import icon from "discourse-common/helpers/d-icon";
+import getURL from "discourse-common/lib/get-url";
+import Logo from "./logo";
+
+export default class HomeLogo extends Component {
+  @service site;
+  @service siteSettings;
+
+  href = getURL("/");
+  session = Session.current();
+  darkModeAvailable = Session.current().darkModeAvailable;
+  title = this.siteSettings.title;
+
+  get showMobileLogo() {
+    return this.site.mobileView && this.logoResolver("mobile_logo").length > 0;
+  }
+
+  get logoUrl() {
+    return this.logoResolver("logo");
+  }
+
+  get logoUrlDark() {
+    return this.logoResolver("logo", { dark: this.darkModeAvailable });
+  }
+
+  get logoSmallUrl() {
+    return this.logoResolver("logo_small");
+  }
+
+  get logoSmallUrlDark() {
+    return this.logoResolver("logo_small", { dark: this.darkModeAvailable });
+  }
+
+  get mobileLogoUrl() {
+    return this.logoResolver("mobile_logo");
+  }
+
+  get mobileLogoUrlDark() {
+    return this.logoResolver("mobile_logo", { dark: this.darkModeAvailable });
+  }
+
+  logoResolver(name, opts = {}) {
+    // get alternative logos for browser dark dark mode switching
+    if (opts.dark) {
+      return this.siteSettings[`site_${name}_dark_url`];
+    }
+
+    // try dark logos first when color scheme is dark
+    // this is independent of browser dark mode
+    // hence the fallback to normal logos
+    if (Session.currentProp("defaultColorSchemeIsDark")) {
+      return (
+        this.siteSettings[`site_${name}_dark_url`] ||
+        this.siteSettings[`site_${name}_url`] ||
+        ""
+      );
+    }
+
+    return this.siteSettings[`site_${name}_url`] || "";
+  }
+
+  @action
+  click(e) {
+    if (wantsNewWindow(e)) {
+      return false;
+    }
+
+    e.preventDefault();
+    DiscourseURL.routeToTag(e.target.closest("a"));
+    return false;
```

Comment:
Is the return value still relevant? It used to be due to jQuery and/or the way ember components handled events but might not be the case anymore.